### ISSUE TITLE
Remove Python 2 Support

### ISSUE THREAD

**Issue by austinmatherne-wk** [Arelle/Arelle#309]
Arelle contains legacy conditional code for running under both Python 2 and 3. Since Python 2 is no longer supported the Python 2 code paths can be removed.

Files with Python 2 conditional code to remove:
* [apidocs/conf.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/apidocs/conf.py)
* [arelle/Cntlr.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/Cntlr.py)
* [arelle/PackageManager.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/PackageManager.py)
* [arelle/PythonUtil.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/PythonUtil.py)
* [arelle/RenderingResolver.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/RenderingResolver.py)
* [arelle/UrlUtil.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/UrlUtil.py)
* [arelle/ViewFile.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/ViewFile.py)
* [arelle/WebCache.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/WebCache.py)
* [arelle/XPathParser.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/XPathParser.py)
* [arelle/plugin/loadFromOIM-2018.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/plugin/loadFromOIM-2018.py)
* [arelle/plugin/saveLoadableOIM.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/plugin/saveLoadableOIM.py)
* [arelle/plugin/sphinx/FormulaGenerator.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/plugin/sphinx/FormulaGenerator.py)
* [arelle/plugin/sphinx/SphinxParser.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/plugin/sphinx/SphinxParser.py)
* [arelle/plugin/validate/USSecTagging.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/plugin/validate/USSecTagging.py)
* [arelle/plugin/xbrlDB/SqlDb.py](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/plugin/xbrlDB/SqlDb.py)

**Comment by ggravlingen** [comment]
@austinmatherne-wk I believe there's some Python 2.7 stuff left in the bottom of FileSource?

**Comment by austinmatherne-wk** [comment]
Do you mean the GAE support? To be honest I'm not sure how/if that's used, or if we need to update support for it or not. I'll need to dig into that at some point and write up tickets to remove or update depending on what makes sense.

**Comment by ggravlingen** [comment]
Yes, exactly. Was I was adding type hints to the file, I came across this https://github.com/Arelle/Arelle/blob/cb10d2365fa4bfeb389f45275b6ac3f3257fb830/arelle/FileSource.py#L767 which [this link](https://cloud.google.com/appengine/docs/legacy/standard/python/refdocs/google.appengine.api) suggests is Python 2.
